src/mantle/support/traits/trait-hookable.php

```diff
@@ -247,4 +268,23 @@ protected function fire_doing_it_wrong( ReflectionMethod $method ): void {
 
 		_doing_it_wrong( esc_html( static::class . '::' . $method->getName() ), esc_html( $message ), '1.0.0' );
 	}
+
+	/**
+	 * Fire all validators for a method.
+	 *
+	 * @param ReflectionMethod $method The hook callback method.
+	 */
+	protected function fire_validator( ReflectionMethod $method ): bool {
+		$attributes = $method->getAttributes();
```

Comment:
this will limit the attributes for you without having to do an instance of below.

```suggestion
		$attributes = $method->getAttributes( \Mantle\Types\Validator::class, \ReflectionAttribute::IS_INSTANCEOF );

		// Check all validators for this method.
		foreach ( $attributes as $attribute ) {
			if ( ! $attribute->newInstance()->validate() ) {
				return false;
			}
```